src/components/checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,82 @@
+import {
+  type Component,
+  createSignal,
+  splitProps,
+  Show,
+  createEffect,
+} from "solid-js";
+import type { JSX } from "solid-js";
+import { checkboxVariants } from "./Checkbox.styles";
+import { type VariantProps, type ClassProps, classes } from "@src/lib/style";
+import type { ComponentProps } from "solid-js";
+
+export type CheckboxProps = VariantProps<typeof checkboxVariants> &
+  ClassProps &
+  ComponentProps<"input"> & {
+    indeterminate?: boolean;
+    label?: JSX.Element;
+  };
+
+const Checkbox: Component<CheckboxProps> = (props) => {
+  const [localProps, variantProps, otherProps] = splitProps(
+    props,
+    [
+      "label",
+      "indeterminate",
+      "type",
+      "onChange",
+      "onFocus",
+      "onBlur",
+      "checked",
+      "disabled",
+    ],
+    ["class", ...checkboxVariants.variantKeys]
+  );
+
+  const [inputRef, setInputRef] = createSignal<HTMLInputElement>();
```

Comment:
I think that you don't need a signal to hold the input ref

```suggestion
  let inputRef: HTMLInputElement | undefined;
```